db/db_impl/db_impl_open.cc

```diff
@@ -1467,25 +1465,29 @@ Status DBImpl::Open(const DBOptions& db_options, const std::string& dbname,
   DBImpl* impl = new DBImpl(db_options, dbname, seq_per_batch, batch_per_txn);
   s = impl->env_->CreateDirIfMissing(impl->immutable_db_options_.wal_dir);
   if (s.ok()) {
-    std::vector<std::string> paths;
+    int32_t pathSize = 0;
     for (auto& db_path : impl->immutable_db_options_.db_paths) {
-      paths.emplace_back(db_path.path);
+      s = impl->env_->CreateDirIfMissing(db_path.path);
+      if (!s.ok()) {
+        delete impl;
+        return s;
+      }
+      pathSize += 1;
     }
     for (auto& cf : column_families) {
       for (auto& cf_path : cf.options.cf_paths) {
-        paths.emplace_back(cf_path.path);
-      }
-    }
-    for (auto& path : paths) {
-      s = impl->env_->CreateDirIfMissing(path);
-      if (!s.ok()) {
-        break;
+        s = impl->env_->CreateDirIfMissing(cf_path.path);
+        if (!s.ok()) {
+          delete impl;
```

Comment:
I like the original code's way, which is to reduce where "return" is called so that we are less likely to forget `delete impl` in the future.